app/services/PageService.scala

```diff
@@ -155,29 +159,86 @@ class PageService @Inject()(config: Configuration, projectRepository: ProjectRep
     } else None
   }
 
-  def computePageFromPath(path: String, refresh: Boolean = false): Future[Option[PageWithContent]] = {
-    val pathWithBranch = projectRepository.findById(path.split(">")(0)).map { project =>
-      if (path.contains(">>")) path.replace(">>", s">${project.stableBranch}>") else path
-    }.getOrElse(s" Error while computing Page for the path $path")
+  def computePageFromPath(path: String, refresh: Boolean = false): Future[Option[PageDTO]] = {
+    val projectOpt = projectRepository.findById(projectIdFromPath(path))
+
+    val pathWithBranch = projectOpt
+      .map { project =>
+        if (path.contains(">>")) {
+          path.replace(">>", s">${project.stableBranch}>")
+        } else {
+          path
+        }
+      }
+      // FIXME is it normal to put an error message in the path variable???
```

Comment:
I'm pretty sure the answer is no!

Reply:
Forgot this one!
Not sure I do it here or part of another PR but will do something about it!